cmsis/device/rtos/CMakeLists.txt

```diff
@@ -35,3 +35,14 @@ target_compile_definitions(mbed-rtos
     INTERFACE
         MBED_CONF_RTOS_PRESENT=1
 )
+
+if(NOT "full" IN_LIST MBED_TARGET_SUPPORTED_APPLICATION_PROFILES)
+   message(FATAL_ERROR
+       "Full profile is not supported for this target")
```

Comment:
What does this mean,  we should use baremetal instead? If yes, we might want to add it here (for a new user it might be helpful hint).